### DISCUSSION TITLE
How can I add "Deliverables" object to the Timewrap of the related "Account"

### DISCUSSION THREAD

**Discussion by StandGitHub** [deejay-hub/timeline-lwc#278] [Q&A] [answered]
I have a question. My team would like to add the Deliverables object to track on the related Account's timeline. Is this even possible as "Deliverables" are not directly linked to Accounts, but it is linked to "Opportunity" which in turn is linked to "Account"
So, for example, the team wants to see all deliverables from all the Opportunities under one Account on the Account's timeline.

Thanks
![Screenshot 2023-12-06 at 3 24 10 PM](https://github.com/deejay-hub/timeline-lwc/assets/67073634/4281a7a4-e605-472e-91f4-f3d5f3faa6f9)

**Comment by deejay-hub** [comment, marked as answer]
The AppExchange package version of Time Warp doesn't support this use case. Traversing relationships to the grand child level creates some challenges to get a query to scale and be performant.  Since in your case we would need to query for all opportunities related to the Account and then using that list of Opportunities use their Ids to then query for all deliverables that are related to any of the Opportunity Ids.

Your best bet would be to take the code and add your use case onto it. Finding a generic way to do this for any Parent - Child - Grandchild relationship in a way that is fast and scales to any use case has eluded me.

**Reply by StandGitHub** [reply]
Thanks for the response. I came to the same conclusion. 

Luckily it turns out, I could just create a lookup relationship between Account and Deliverables (either make the lookup field mandatory so people would fill it when they create deliverables or have a flow fill out based on opportunity's Account)
This would make the relationship a Parent-Child instead of a grandparent, and the documentation (https://quip.com/ahFCA9VBEKtr) covers this use case for the custom object from a managed package.

 I tested it in sandbox and this solved my use case to have all the deliverables from Opportunities show up on the corresponding Account's timeline
